sslutils.c

```diff
@@ -844,7 +859,11 @@ openssl_csr_to_crt(PG_FUNCTION_ARGS)
 		goto out;
 	}
 
-	if (!X509_gmtime_adj(X509_get0_notAfter(certificate), (long)60 * 60 * 24 * VALIDITY_DAYS))
+	if (!PG_ARGISNULL(3))
+	{
+		validity_days = PG_GETARG_INT32(3);
+	}
```

Comment:
Validity must be `> 0` (It must not be less than or equal to 0).
Please add validation.